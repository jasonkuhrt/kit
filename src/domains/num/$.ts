/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
