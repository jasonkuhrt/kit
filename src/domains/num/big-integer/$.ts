export * as BigInteger from './$$.ts'
