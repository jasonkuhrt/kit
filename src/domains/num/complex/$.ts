export * as Complex from './$$.js'
