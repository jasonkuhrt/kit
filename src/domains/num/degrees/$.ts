// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
