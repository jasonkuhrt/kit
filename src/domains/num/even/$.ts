// Export the namespace containing all Even operations
export * as Even from './$$.ts'
