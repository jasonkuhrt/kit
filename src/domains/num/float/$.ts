// Export the namespace containing all Float operations
export * as Float from './$$.js'
