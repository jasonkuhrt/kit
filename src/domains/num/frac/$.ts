// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
