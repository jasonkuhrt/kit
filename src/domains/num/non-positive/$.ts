// Export the namespace containing all NonPositive operations
export * as NonPositive from './$$.js'
