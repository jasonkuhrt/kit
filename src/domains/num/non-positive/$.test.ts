import { Test } from '#test'
import { NonPositive } from './$.js'

Test.describe('is')
  .on(NonPositive.is)
  .casesAsArgs(
    // Valid cases - non-positive numbers (<= 0)
    0,
    -1,
    -100,
    -0.5,
    -Number.MAX_VALUE,
    -Infinity,
    // Invalid cases - positive numbers
    1,
    0.1,
    100,
    Infinity,
    NaN,
    '0',
    null,
  )
  .test()

Test.describe('tryFrom')
  .on(NonPositive.tryFrom)
  .casesAsArgs(
    // Valid cases
    0,
    -1,
    -100,
    -0.5,
    -Number.MAX_VALUE,
    -Infinity,
    // Invalid cases - should return null
    1,
    0.1,
    100,
    Infinity,
    NaN,
  )
  .test()
