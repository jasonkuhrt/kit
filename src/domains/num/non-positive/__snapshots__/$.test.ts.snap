// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`is > is("0") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1.7976931348623157e+308) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.7976931348623157e+308
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-0.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1.7976931348623157e+308) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.7976931348623157e+308
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-1.7976931348623157e+308
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-100
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;
